File: edgevox-onnx/python/csrc/offline-speech-denoiser-gtcrn-model-config.cc
// edgevox-onnx/python/csrc/offline-speech-denoiser-gtcrn-model-config.cc
//
// Copyright (c)  2025  Xiaomi Corporation

#include "edgevox-onnx/python/csrc/offline-speech-denoiser-gtcrn-model-config.h"

#include <string>

#include "edgevox-onnx/csrc/offline-speech-denoiser-gtcrn-model-config.h"

namespace edgevox_onnx {

void PybindOfflineSpeechDenoiserGtcrnModelConfig(py::module *m) {
  using PyClass = OfflineSpeechDenoiserGtcrnModelConfig;
  py::class_<PyClass>(*m, "OfflineSpeechDenoiserGtcrnModelConfig")
      .def(py::init<const std::string &>(), py::arg("model") = "")
      .def_readwrite("model", &PyClass::model)
      .def("validate", &PyClass::Validate)
      .def("__str__", &PyClass::ToString);
}

}  // namespace edgevox_onnx

File: edgevox-onnx/java-api/src/main/java/com/nexus/edgevox/onnx/OfflineSpeechDenoiserGtcrnModelConfig.java
// Copyright 2025 Xiaomi Corporation
package com.nexus.edgevox.onnx;

public class OfflineSpeechDenoiserGtcrnModelConfig {
    private final String model;

    private OfflineSpeechDenoiserGtcrnModelConfig(Builder builder) {
        this.model = builder.model;
    }

    public static Builder builder() {
        return new Builder();
    }

    public String getModel() {
        return model;
    }

    public static class Builder {
        private String model = "";

        public OfflineSpeechDenoiserGtcrnModelConfig build() {
            return new OfflineSpeechDenoiserGtcrnModelConfig(this);
        }

        public Builder setModel(String model) {
            this.model = model;
            return this;
        }
    }
}
